redbot/core/bot.py

```diff
@@ -203,47 +229,112 @@ def __init__(self, *args, cli_flags=None, bot_dir: Path = Path.cwd(), **kwargs):
+            for snowflake in await self._config.guild(member.guild).admin_role():
                 if member_snowflakes.has(snowflake):  # Dep-WARN
                     return True
         except AttributeError:  # someone passed a webhook to this
             pass
         return False
 
-    async def is_mod(self, member: discord.Member):
+    async def is_mod(self, member: discord.Member) -> bool:
         """Checks if a member is a mod or admin of their guild."""
         try:
             member_snowflakes = member._roles  # DEP-WARN
-            for snowflake in await self.db.guild(member.guild).admin_role():
+            for snowflake in await self._config.guild(member.guild).admin_role():
                 if member_snowflakes.has(snowflake):  # DEP-WARN
                     return True
-            for snowflake in await self.db.guild(member.guild).mod_role():
+            for snowflake in await self._config.guild(member.guild).mod_role():
                 if member_snowflakes.has(snowflake):  # DEP-WARN
                     return True
         except AttributeError:  # someone passed a webhook to this
             pass
         return False
 
+    async def get_admin_roles(self, guild: discord.Guild) -> List[discord.Role]:
+        """
+        Gets the admin roles for a guild.
+        """
+        ret: List[discord.Role] = []
+        for snowflake in await self._config.guild(guild).admin_role():
+            r = guild.get_role(snowflake)
+            if r:
+                ret.append(r)
+        return ret
+
+    async def get_mod_roles(self, guild: discord.Guild) -> List[discord.Role]:
+        """
+        Gets the mod roles for a guild.
+        """
+        ret: List[discord.Role] = []
+        for snowflake in await self._config.guild(guild).mod_role():
+            r = guild.get_role(snowflake)
+            if r:
+                ret.append(r)
+        return ret
+
+    async def get_admin_role_ids(self, guild_id: int) -> List[int]:
+        """
+        Gets the admin role ids for a guild id.
+        """
+        return await self._config.guild(discord.Object(id=guild_id)).admin_role()
+
+    async def get_mod_role_ids(self, guild_id: int) -> List[int]:
+        """
+        Gets the mod role ids for a guild id.
+        """
+        return await self._config.guild(discord.Object(id=guild_id)).mod_role()
+
+    async def get_shared_api_tokens(self, service_name: str) -> Dict[str, str]:
+        """
+        Gets the shared API tokens for a service
+
+        Parameters
+        ----------
+        service_name: str
+
+        Returns
+        -------
+        Dict[str, str]
+            A Mapping of token names to tokens.
+            This mapping exists because some services have multiple tokens.
+        """
+        return await self._config.api_tokens.get_raw(service_name, default={})
+
+    async def set_shared_api_tokens(self, service_name: str, **tokens: str):
```

Comment:
This seems to always throw `AttributeError: 'Value' object has no attribute 'tokens'` for line 335
```py
Traceback (most recent call last):
  File "<venv>\lib\site-packages\redbot\core\dev_commands.py", line 182, in _eval
    result = await func()
  File "<string>", line 2, in func
  File "<venv>\lib\site-packages\redbot\core\bot.py", line 335, in set_shared_api_tokens
    async with self._config.api.tokens.get_attr(service_name)() as method_abuse:
AttributeError: 'Value' object has no attribute 'tokens'
```

Reply:
Great. (fixed)